driver/src/main/com/mongodb/MongoClient.java

```diff
@@ -368,8 +369,25 @@ public MongoClientOptions getMongoClientOptions() {
      * @since 3.0
      */
     public MongoIterable<String> listDatabaseNames() {
-        return new ListDatabasesIterableImpl<BsonDocument>(BsonDocument.class, getDefaultCodecRegistry(),
-                ReadPreference.primary(), createOperationExecutor()).map(new Function<BsonDocument, String>() {
+        return executeListDatabaseNames(null);
+    }
+
+    /**
+     * Get a list of the database names
+     *
+     * @param clientSession the client session with which to associate this operation
+     * @return an iterable containing all the names of all the databases
+     * @since 3.6
+     * @mongodb.server.release 3.6
+     * @mongodb.driver.manual reference/command/listDatabases List Databases
+     */
+    public MongoIterable<String> listDatabaseNames(final ClientSession clientSession) {
+        return executeListDatabaseNames(clientSession);
+    }
+
+    private MongoIterable<String> executeListDatabaseNames(final ClientSession clientSession) {
+        notNull("clientSession", clientSession);
```

Comment:
Can't have a notNull check here - doesn't the ` listDatabaseNames()` method calls it with null?

Reply:
Indeed.  This method is not unit tested because of the way it uses map.  I'll figure out how to test.